viral/offline_sequence_detection.py

```
@@ -0,0 +1,940 @@
+
+    Essentially, this is a Python implementation of https://github.com/losonczylab/Grosmark_NatNeuro_2021/blob/main/shufflePopulationEvents.m
+    """
+    # as we don't have a circular run belt but a linear corridor, we are using the weighted linear correlation coefficients
+    xy = construct_xy_by_bin(posterior_probability_matrix, mode=mode)
+    if mode == "linear":
+        weighted_r = calculate_linear_weighted_correlation(
+            posterior_probability_matrix, xy
+        )
+        print(f"Linear weighted correlation: {weighted_r}")
+        shuffled_weighted_rs = list()
+        for i in range(n_shuffles):
```

Ruff (0.14.5)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)